src/main/java/org/opensearch/sdk/ExtensionRestRequest.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.sdk;
+
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.extensions.rest.RestExecuteOnExtensionRequest;
+import org.opensearch.identity.PrincipalIdentifierToken;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.RestRequest.Method;
+import org.opensearch.transport.TransportRequest;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class ExtensionRestRequest extends TransportRequest {
+
+    private Method method;
+    private String uri;
+    private PrincipalIdentifierToken principalIdentifierToken;
+
+    public ExtensionRestRequest(Method method, String uri, PrincipalIdentifierToken requesterIdentifier) {
+        this.method = method;
+        this.uri = uri;
+        this.principalIdentifierToken = requesterIdentifier;
+    }
+
+    // TODO Check if this constructor is more useful, If so what happens when request object is null
+    protected ExtensionRestRequest(RestExecuteOnExtensionRequest request) {
+        this.method = request.getMethod();
+        this.uri = request.getUri();
+        this.principalIdentifierToken = request.getRequestIssuerIdentity();
+    }
+
+    public ExtensionRestRequest(StreamInput in) throws IOException {
+        super(in);
+        try {
+            method = RestRequest.Method.valueOf(in.readString());
```

Comment:
Thanks for creating this new class. I like it.

When I wrote this earlier but later discovered the StreamInput has enum-writing methods.   So this whole try/catch block could just be 
```java
method = in.readEnum(RestRequest.Method);`
```